nixos/modules/services/networking/seafile.nix

```diff
@@ -0,0 +1,229 @@
+      type = types.string;
+      description = ''
+        Seafile Seahub Admin Account Email.
+      '';
+    };
+
+    initialAdminPassword = mkOption {
+      default = "seafile_password";
+      example = "123456";
+      type = types.string;
+      description = ''
+        Seafile Seahub Admin Account initial password.
+        Should be change via Seahub web front-end.
+      '';
+    };
+
+    seahubExtraConf = mkOption {
+      default = "";
+      type = types.lines;
+      description = ''
+        Extra config to append to `seahub_settings.py` file.
+      '';
+    };
+  };
+
+  ###### Implementation
+
+  config = mkIf cfg.enable {
+
+    systemd.services.ccnet = {
+      description = "Seafile ccnet Server";
+      after = [ "network.target" ];
+      serviceConfig = {
+        Type = "forking";
+        User = "seafile";
+        Group = "seafile";
+        RuntimeDirectory = "seafile";
+        ExecStart = "${pkgs.ccnet}/bin/ccnet-server -F ${centralConfigDir} -c ${ccnetDir} -f ${logDir}/ccnet.log -d -P ${runDir}/ccnet.pid";
+        PIDFile = "${runDir}/ccnet.pid";
+      };
+      preStart = ''
+        mkdir -p ${logDir}
+        if [ ! -d "${ccnetDir}" ]; then
+          ${pkgs.ccnet}/bin/ccnet-init -F ${centralConfigDir} -c ${ccnetDir} -n ${cfg.serverName} -H ${cfg.ccnetHost} -P ${toString cfg.ccnetPort}
+          sed -i "s|^SERVICE_URL.*|SERVICE_URL = ${cfg.serviceUrl}|" ${centralConfigDir}/ccnet.conf
+          ln -s ${seafileIni} ${ccnetDir}/seafile.ini
+        fi
+      '';
+    };
+
+    systemd.services.seafile = {
+      description = "Seafile Server";
+      after = [ "network.target" "ccnet.service" ];
+      requires = [ "ccnet.service" ];
+      serviceConfig = {
+        Type = "forking";
+        User = "seafile";
+        Group = "seafile";
+        RuntimeDirectory = "seafile";
+        ExecStart = "${pkgs.seafile-server}/bin/seaf-server -F ${centralConfigDir} -c ${ccnetDir} -d ${seafileDataDir} -l ${logDir}/seafile.log -P ${runDir}/seafile.pid";
+        PIDFile = "${runDir}/seafile.pid";
+      };
+      preStart = ''
+        if [ ! -d "${seafileDataDir}" ]; then
+          ${pkgs.seafile-server}/bin/seaf-server-init -F ${centralConfigDir} \
+            --seafile-dir ${seafileDataDir} --port ${toString cfg.seafilePort} --fileserver-port ${toString cfg.fileserverPort}
+        fi
+      '';
+    };
+
+    systemd.services.seahub = {
+      description = "Seafile Server Web Frontend";
+      wantedBy = [ "multi-user.target" ];
+      after = [ "network.target" "seafile.service" ];
+      requires = [ "seafile.service" ];
+      environment = {
+        CCNET_CONF_DIR = ccnetDir;
+        SEAFILE_CONF_DIR = seafileDataDir;
+        SEAFILE_CENTRAL_CONF_DIR = centralConfigDir;
+      };
+      serviceConfig = {
+        Type = "forking";
+        User = "seafile";
+        Group = "seafile";
+        RuntimeDirectory = "seafile";
+        WorkingDirectory = pkgs.seafile-seahub;
```

Comment:
Seems a little fragile to me, would it make sense to instead set it to the data dir? If the service expects the files in the store path to be in cwd, they could be copied at `preStart`.